Prog/NCHousing_needs_units_targets.sas

```diff
@@ -0,0 +1,927 @@
+	    Using 4.1% as the effective mortgage rate for NC in 2017, 
+	    calculate monthly P & I payment using monthly mortgage rate and compounded interest calculation
+	    ******; 
+	    
+	    loan = .9 * valueh_a;
+	    month_mortgage= (4.1 / 12) / 100; 
+	    monthly_PI = loan * month_mortgage * ((1+month_mortgage)**360)/(((1+month_mortgage)**360)-1);
+
+	    ****
+	    Calculate PMI and taxes/insurance to add to Monthly_PI to find total monthly payment
+	    ******;
+	    
+	    PMI = (.007 * loan ) / 12; **typical annual PMI is .007 of loan amount;
+	    tax_ins = .25 * monthly_PI; **taxes assumed to be 25% of monthly PI; 
+	    total_month = monthly_PI + PMI + tax_ins; **Sum of monthly payment components;
+
+		
+	
+		*owner cost categories that make more sense for owner costs - no longer used in targets;
+       /*need to discuss for NC*/
+		ownlevel=.;
+			if 0 <=total_month<1200 then ownlevel=1;
+			if 1200 <=total_month<1800 then ownlevel=2;
+			if 1800 <=total_month<2500 then ownlevel=3;
+			if 2500 <=total_month<3200 then ownlevel=4;
+			if 3200 <=total_month<4200 then ownlevel=5;
+			if total_month >= 4200 then ownlevel=6;
+
+		mownlevel=.;
+			if max_ocost<1200 then mownlevel=1;
+			if 1200 <=max_ocost<1800 then mownlevel=2;
+			if 1800 <=max_ocost<2500 then mownlevel=3;
+			if 2500 <=max_ocost<3200 then mownlevel=4;
+			if 3200 <=max_ocost<4200 then mownlevel=5;
+			if max_ocost >= 4200 then mownlevel=6;
+
+        *Leah: this is where it differs from the other program
+		 		*owner cost categories now used in targets that provide a set of categories useable for renters and owners combined; 
+			allcostlevel=.;
+			if total_month<800 then allcostlevel=1;
+			if 800 <=total_month<1300 then allcostlevel=2;
+			if 1300 <=total_month<1800 then allcostlevel=3;
+			if 1800 <=total_month<2500 then allcostlevel=4;
+			if 2500 <=total_month<3500 then allcostlevel=5;
+			if total_month >= 3500 then allcostlevel=6; 
+
+				mallcostlevel=.;
+			if max_ocost<800 then mallcostlevel=1;
+			if 800 <=max_ocost<1300 then mallcostlevel=2;
+			if 1300 <=max_ocost<1800 then mallcostlevel=3;
+			if 1800 <=max_ocost<2500 then mallcostlevel=4;
+			if 2500 <=max_ocost<3500 then mallcostlevel=5;
+			if max_ocost >= 3500 then mallcostlevel=6;
+
+  end;
+
+  *add structure of housing variable;
+  if UNITSSTR = 00 then structure=5;
+if UNITSSTR in (01, 02) then structure=4;
+if UNITSSTR in (03, 04) then structure=1;
+if UNITSSTR in (05, 06, 07) then structure=2;
+if UNITSSTR in (08, 09, 10) then structure=3;
+	
+	if BUILTYR2 in ( 00, 9999999, .n , . ) then structureyear=.;
+		else do; 
+		    if BUILTYR2  in (07, 08, 09, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22) then structureyear=1;
+			else if BUILTYR2  in (04, 05, 06) then structureyear=2;
+            else if BUILTYR2 in (01, 02, 03)  then structureyear=3;
+		end;
+
+		
+	if rentgrs in ( 9999999, .n , . ) then affordable=.;
+		else do; 
+		    if rentgrs_a<750 then affordable=1;
```

Comment:
this may need to be adjusted.

Reply:
According to the new costlevel cutoffs are 350, 700, 1050, 1500, 2500, this should be 350? @lhendey

Reply:
@maggiesu0725 no - we didn't use the bottom category the last time - we used the second - (which was $1300) so use $700?